x-pack/plugins/fleet/server/services/epm/packages/cleanup.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SavedObjectsClientContract } from 'src/core/server';
+
+import { removeArchiveEntries } from '../archive/storage';
+
+import { ASSETS_SAVED_OBJECT_TYPE, PACKAGE_POLICY_SAVED_OBJECT_TYPE } from '../../../../common';
+import type { PackageAssetReference } from '../../../../common';
+import { packagePolicyService } from '../../package_policy';
+import { appContextService } from '../..';
+
+export async function removeOldAssets(options: {
+  savedObjectsClient: SavedObjectsClientContract;
+  pkgName: string;
+  oldVersion: string;
+}) {
+  const { savedObjectsClient, pkgName, oldVersion } = options;
+
+  // check if old version has assets
+  const assets = await savedObjectsClient.find<PackageAssetReference>({
+    type: ASSETS_SAVED_OBJECT_TYPE,
+    filter: `${ASSETS_SAVED_OBJECT_TYPE}.attributes.package_name:${pkgName} AND ${ASSETS_SAVED_OBJECT_TYPE}.attributes.package_version:${oldVersion}`,
+    // how to query if more than one page?
+    perPage: 100,
```

Comment:
I recommend using the `savedObjectsClient.createPointInTimeFinder` utility for paging which will give you a stable "cursor" of sorts that can be used in conjunction with the `find` API.

Reply:
thanks, I'll check this

Reply:
I rewrote this to use point in time finder